src/GafferUI/StandardGraphLayout.cpp

```diff
@@ -526,9 +527,17 @@ class LayoutEngine
 			for( VertexIterator it = v.first; it != v.second; ++it )
 			{
 				const Vertex &v = m_graph[*it];
-				if( v.node )
+				if( !v.pinned && v.node )
 				{
-					m_graphGadget->setNodePosition( v.node, v.position );
+					if( !isnan( v.position.x ) && !isnan( v.position.y ) && !isinf( v.position.x ) && !isinf( v.position.y ) )
+					{
+						m_graphGadget->setNodePosition( v.node, v.position );
+					}
+					else
+					{
+						IECore::msg( IECore::Msg::Warning, "LayoutEngine::computeAttributes", "Layout algorithm failed to produce valid position." );
```

Comment:
I suspect the reference to computeAttributes has been copy pasted from somewhere else? Could we fix that and also include the the node name in the message?